### DISCUSSION TITLE
Add an ant target at the end of an existing transform type

### DISCUSSION THREAD

**Discussion by gdelory** [dita-ot/dita-ot#4531] [Q&A] [answered]
Hi all,

I'm trying to run an Ant target at the end of the `html5` transform. It doesn't seem to be an extension point [at the end of the target](https://github.com/dita-ot/dita-ot/blob/4.2.3/src/main/plugins/org.dita.html5/build_dita2html5_template.xml#L14-L20), and the [depend.preprocess.post](https://www.dita-ot.org/dev/extension-points/plugin-extension-points-pre-processing#depend.preprocess.post) extension point is too early for me (I'm trying to move files around in the output folder at the end of the transform, to end up only with in-scope content, as a temporary work-around for [this discussion](https://github.com/orgs/dita-ot/discussions/4522)).

I know I can easily [create my own new transform type](https://www.dita-ot.org/dev/topics/plugin-newtranstype) and depends on the `html5` transform then my new ant target doing the deleting/moving of the files, but the issue it that we already have thousands of transforms/users using the `html5` transform and we would rather avoid changing the name of the transform.

The best I found so far would be to import my ant file with [ant.import](https://www.dita-ot.org/dev/topics/plugin-anttarget) (same as for a new transform anyway), and override the `dita2html5` target. The tricky part was to reference the original `dita2html5` target, since my new target has the same name. I managed by prefixing the original target with the project name though:

```xml
<target name="dita2html5"
  depends="dita2html5.dita2html5,
  no-outer-content.move"/>
```

My question is: Is there a better way to run an Ant target at the end of an existing transform, like an extension point I missed, or something else?

Thank you

**Comment by raducoravu** [comment, marked as answer]
Looking at this template:

DITA-OT/plugins/org.dita.base/build_template.xml

    <target name="init">
    <dita-ot-fail id="DOTA001F">
      <condition>
        <and>
          <dita:extension id="dita.conductor.transtype.check" behavior="org.dita.dost.platform.CheckTranstypeAction"
                          property="transtype"/>
        </and>
      </condition>
      <param name="1" value="${transtype}"/>
    </dita-ot-fail>
    <property name="clean.temp" value="yes"/>
    <property name="build-step.clean-temp" value="${clean.temp}"/>
    <preprocess-skip-init name="clean-temp" step="clean-temp"/>
    <antcall inheritRefs="true">
      <target name="dita2${transtype}"/>
      <target name="clean-temp"/>
    </antcall>
  </target>

The "clean-temp" seems to be done last.
And looking at the definition of the "clean-temp" target in "DITA-OT/plugins/org.dita.base/build_preprocess_template.xml" it seems to depend on this extension point "depend.preprocess.clean-temp.pre".

**Reply by gdelory** [reply]
Thanks @raducoravu that's an excellent idea! I actually initially saw that `clean-temp` was run after the main `dita2${transtype}` target, but didn't think at all to go check if it had an extension point 😄 

The good news is that even when you have `-Dclean.temp=no`, the `depend.preprocess.clean-temp.pre` extension point seems to run, I just tried to make sure. We don't really use it on production obviously, but I use it a lot when developing so it's nice that it still runs even when the temp clean up is disabled.

I like it more that overwriting the `dita2html5` target which feels a bit hacky, so I probably will go with your solution. Thanks again.
